### PR TITLE
issue #1412 re-connecting to postgres when connection is lost

### PR DESCRIPTION
**Pull request by cfreyfh**
## Proposed changes

It fixes that the PostgreSQL connections where not re-connected once they are closed by the PostgreSQL server (e.q. restart of the server or network problems)

## Types of changes

What types of changes does your code introduce to the project: _Put an `x` in the boxes that apply_

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before
merging your code._

-   [x] I have read the [CONTRIBUTING](https://github.com/FIWARE/context.Orion-LD/blob/master/CONTRIBUTING.md) doc
-   [ ] I have signed the [CLA](https://fiware.github.io/contribution-requirements/individual-cla.pdf)
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)
-   [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you
did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by cfreyfh** [comment]
refering to issue https://github.com/FIWARE/context.Orion-LD/issues/1412

File: src/lib/orionld/troe/pgConnectionGet.cpp [hunk 1]
```diff
@@ -92,22 +92,44 @@ PgConnection* pgConnectionGet(const char* db)
...
+
+              // if still no connection
+              if (pgStatus != CONNECTION_OK)
+              {
+                  // we free this pointer that it can be used in the next call of pgConnectionGet
+                  free(poolP->connectionV[ix]);
+                  poolP->connectionV[ix] = NULL;
+                  LM_W(("Connection failed, pointer of item %d was re-set to NULL (%p)", ix, poolP->connectionV[ix]));
+                  // this time no success finding a connection that is working, try in the next loop
+                  continue;
```
Comment (kzangeli) [inline src/lib/orionld/troe/pgConnectionGet.cpp]: You seem to have changed the indentation to 4 spaces.
I'd really prefer to have two spaces, as in the rest of the code base.

**Comment by kzangeli** [comment]
Thank you for contributing!
github asks you to "sign the Contributor License Agreement", by just addding a comment with the text:
> I have read the CLA Document and I hereby sign the CLA

File: src/lib/orionld/troe/pgConnectionGet.cpp (hunk 1 above)
Comment (cfreyfh) [reply src/lib/orionld/troe/pgConnectionGet.cpp]: I changed the indentation to 2 spaces.

**Comment by cfreyfh** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by kzangeli** [comment]
Thank You very much for contributing!

**Pull request merged**